webapp/src/js/services/enketo.js

```diff
@@ -574,13 +574,29 @@ angular.module('inboxServices').service('Enketo',
           return xmlToDocs(doc, form.getDataStr({ irrelevant: false }));
         })
         .then(function(docs) {
-          if (geolocation) {
-            // Only update geolocation if one is provided.  Otherwise, maintain
-            // whatever is already set in the docs.
-            docs.forEach(function(doc) {
-              doc.geolocation = geolocation;
-            });
+          if (geoHandle) {
+            return geoHandle()
+            // REVIEWER
+            //   I'm not sure about this wrt edits. As it is written:
+            //    A) Create with no geo: no geolocation with relevant error
+            //    B) Create with geo: relevant geolocation with no error
+            //    C) Edit A with geo: relevant geolocation with old error
+            //    D) Edit A with no geo: no geolocation with relevant error
+            //    E) Edit B with no geo: old geolocation with relevant error
+            //    F) Edit B with geo: new geolcation with blank error
+            //
+            //  Perhaps we should just always overwrite doc.geolocation with either the success or the failure?
+            //  This is not what master does right now though, master follows the above but doesn't store the error
+            // REVIEWER
+              .then(coords => docs.forEach(doc => doc.geolocation = coords))
+              .catch(error => docs.forEach(doc => doc.geolocation_error = error))
```

Comment:
It would be too confusing to have both geo and geo_error set IMO... I think what I mean is...

A) no geolocation with relevant error
B) geolocation with no error
C) geolocation with no error (the edit wipes the error out)
D) no geolocation with latest error
E) old geolocation with no error
F) latest geolocation with no error

So you only get an error if every time you've saved the doc you got an error.

Reply:
@abbyad do you have a perspective here? This is important for data science people if we care about edits, as what Gareth is proposing will mean that on edited documents the geo may or may not be located and where the edit occurred.

Reply:
Edits are very tricky here. I would think that we should either:
1) never change geolocation on edit, or
2) have a log of the geolocations for every save, along with timestamp.

Overwriting the original geolocation seems problematic to me since the location data is not as reliable. 

Curious what @isaacholeman or @ecsalomon would say from a research perspective.

Reply:
Cross-referencing a response: https://github.com/medic/cht-core/pull/6535#issuecomment-659689435

Reply:
Moving this discussion to the main thread